src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -267,7 +269,14 @@ create_reference_table(PG_FUNCTION_ARGS)
 	 * sense of this table until we've committed, and we don't want multiple
 	 * backends manipulating this relation.
 	 */
-	Relation relation = relation_open(relationId, ExclusiveLock);
+	Relation relation = try_relation_open(relationId, ExclusiveLock);
+	if (relation == NULL)
+	{
+		ereport(ERROR, (errmsg("could not create reference table: "
```

Comment:
can't easily test this as normally `create_reference_table` already checks if relation exists